Bug: Enum values cannot be used directly in arithmetic expressions

## Description

Enum values that resolve to integers cannot be used directly in arithmetic expressions. The type checker rejects the operation even though the underlying value is an integer.

## Steps to Reproduce

```ez
import @std
using std

const Direction enum {
    NORTH   // 0
    EAST    // 1
    SOUTH   // 2
    WEST    // 3
}

do main() {
    // This fails:
    temp next_dir int = Direction.NORTH + 1
    
    // This works:
    temp dir int = Direction.NORTH
    temp next_dir2 int = dir + 1
}
```

## Expected Behavior

Since `Direction.NORTH` evaluates to `0` (an integer), the expression `Direction.NORTH + 1` should be valid and evaluate to `1`.

## Actual Behavior

```
error[E3002]: invalid operands for '+': Direction and int (expected numeric or string)
  --> test.ez:13:34
   |
13 |     temp next_dir int = Direction.NORTH + 1
   |                                  ^ operator not valid for type
```

## Workaround

Store the enum value in an int variable first:
```ez
temp dir int = Direction.NORTH
temp next_dir int = dir + 1
```

## Discussion

This could be intentional to enforce type safety with enums. If so, consider:
1. Adding documentation about this behavior
2. Providing an explicit conversion function like `int(Direction.NORTH)`

If not intentional, the type checker should recognize that integer enums are compatible with arithmetic operations.

--- a/pkg/typechecker/typechecker.go
+++ b/pkg/typechecker/typechecker.go
@@ -6876,6 +6876,9 @@ func (tc *TypeChecker) isNumericType(typeName string) bool {
 		"float", "f32", "f64", "byte":
 		return true
 	default:
+		if t, exists := tc.types[typeName]; exists && t.Kind == EnumType {
+			return t.EnumBaseType == "int" || t.EnumBaseType == "float"
+		}
 		return false
 	}
 }

